var/spack/repos/builtin/packages/rocsolver/package.py

```diff
@@ -26,21 +27,25 @@ class Rocsolver(CMakePackage):
     depends_on('cmake@3:', type='build')
     depends_on('numactl', type='link', when='@3.7.0:')
 
-    for ver in ['3.7.0', '3.8.0', '3.9.0', '3.10.0', '4.0.0']:
+    for ver in ['3.7.0', '3.8.0', '3.9.0', '3.10.0', '4.0.0', '4.1.0']:
         depends_on('hsa-rocr-dev@' + ver, type='build', when='@' + ver)
 
-    for ver in ['3.5.0', '3.7.0', '3.8.0', '3.9.0', '3.10.0', '4.0.0']:
+    for ver in ['3.5.0', '3.7.0', '3.8.0', '3.9.0', '3.10.0', '4.0.0', '4.1.0']:
         depends_on('hip@' + ver, type='build', when='@' + ver)
         depends_on('rocm-device-libs@' + ver, type='build', when='@' + ver)
         depends_on('comgr@' + ver, type='build', when='@' + ver)
         depends_on('rocblas@' + ver, type='link', when='@' + ver)
 
     def cmake_args(self):
+        incl = self.spec['rocblas'].prefix
         args = [
             '-DBUILD_CLIENTS_SAMPLES=OFF',
             '-DBUILD_CLIENTS_TESTS=OFF',
             '-DBUILD_CLIENTS_BENCHMARKS=OFF'
         ]
+        if self.spec.satisfies('@4.1.0:'):
```

Comment:
Why is this necessary? It shouldn't be?

Reply:
I have included as for rocm-4.1.0, in compiling this file , /tmp/root/spack-stage/spack-stage-rocsolver-4.1.0-ifnhpjic5mvqbph7y7wz3utudmoaiace/spack-src/rocsolver/library/src/auxiliary/rocauxiliary_orgtr_ungtr.cpp , i see the below error ...

In file included from /tmp/root/spack-stage/spack-stage-rocsolver-4.1.0-ifnhpjic5mvqbph7y7wz3utudmoaiace/spack-src/rocsolver/library/src/buildinfo.cpp:5:
In file included from /tmp/root/spack-stage/spack-stage-rocsolver-4.1.0-ifnhpjic5mvqbph7y7wz3utudmoaiace/spack-src/rocsolver/library/src/include/rocblas.hpp:10:
In file included from /tmp/root/spack-stage/spack-stage-rocsolver-4.1.0-ifnhpjic5mvqbph7y7wz3utudmoaiace/spack-src/rocsolver/library/src/include/common_device_helpers.hpp:8:
In file included from /tmp/root/spack-stage/spack-stage-rocsolver-4.1.0-ifnhpjic5mvqbph7y7wz3utudmoaiace/spack-src/rocsolver/library/src/include/libcommon.hpp:7:
/tmp/root/spack-stage/spack-stage-rocsolver-4.1.0-ifnhpjic5mvqbph7y7wz3utudmoaiace/spack-src/rocsolver/library/src/include/../../../../rocblascommon/library/src/include/utility.h:8:10: fatal error: 'rocblas.h' file not found #include "rocblas.h"

Reply:
did you make an issue for this?

Reply:
[rocsolver_4.1.0_build_error.txt](https://github.com/spack/spack/files/6244034/rocsolver_4.1.0_build_error.txt)
No, Not yet.. I have attached the full error text for your reference..

Reply:
okay, that's weird, i don't see rocblas being among the include flags anywhere... might be worth looking at the CMakeCache.txt file (e.g. use `spack install --keep-stage rocsolver` or `spack stage rocsolver` `spack cd --stage rocsolver` `spack build-env rocsolver -- /bin/bash` `mkdir build` `cd build` `cmake ..`)

Reply:
[rocsolver_4.1.0_CMakeCache.txt](https://github.com/spack/spack/files/6245899/rocsolver_4.1.0_CMakeCache.txt)
Attached is the CMakeCache.txt with out the -DCMAKE_CXX_FLAGS:String=-I{0}/rocblas/include'.format(incl) change.